packages/components/form-field/src/FormField.test.tsx

```diff
@@ -246,4 +246,39 @@ describe('FormField error message', () => {
     expect(inputEl).toHaveValue('Hello')
     expect(charsCountEl).toBeVisible()
   })
+
+  it('should render the characters count component properly when `value` is undefined', () => {
+    const MAX_LENGTH = 6
+    const DEFAULT_VALUE = undefined
+
+    render(
+      <FormField name="text">
+        <FormField.Label>Text</FormField.Label>
+
+        <FormField.Control>
+          {({ id, name }) => (
+            <input
+              type="text"
+              id={id}
+              name={name}
+              defaultValue={DEFAULT_VALUE}
+              maxLength={MAX_LENGTH}
+            />
+          )}
+        </FormField.Control>
+
+        <div className="flex justify-between gap-md">
+          <FormField.HelperMessage>We will never share your email</FormField.HelperMessage>
+
+          <FormField.CharactersCount value={DEFAULT_VALUE} maxLength={MAX_LENGTH} />
+        </div>
+      </FormField>
+    )
+
+    const inputEl = screen.getByLabelText('Text')
+    const charsCountEl = screen.getByText('0/6')
+
+    expect(inputEl).toHaveValue('')
+    expect(charsCountEl).toBeVisible()
```

Comment:
Personal opinion but I think it is easier to read assertions without variables

```
expect(screen.getByLabelText('Text')).toHaveValue('')
expect(screen.getByText('0/6')).toBeVisible()
```

Reply:
Agree, I was based on a different example where it makes more sense, not here ;)